runtime/src/cost_tracker.rs

```diff
@@ -29,6 +31,22 @@ pub enum CostTrackerError {
     WouldExceedAccountDataTotalLimit,
 }
 
+impl From<CostTrackerError> for TransactionError {
```

Comment:
nit: Might be nice to have this in its' own PR to keep the feature-gate PR as slim as possible, but I guess we might run into unused code warning if we do that?

Reply:
Like this idea, https://github.com/solana-labs/solana/pull/30267 for this, I will rebase once that's merged.